module/applications/journal/class-summary-sheet.mjs

```diff
@@ -0,0 +1,439 @@
+
+  /**
+   * Prepare features granted by various advancement types.
+   * @param {Item5e} item  Class item belonging to this journal.
+   * @returns {object}     Prepared advancement section.
+   */
+  _getAdvancement(item) {
+    const advancement = {};
+
+    const hp = item.advancement.byType.HitPoints[0];
+    if ( hp ) {
+      advancement.hp = {
+        hitDice: `1${hp.hitDie}`,
+        max: hp.hitDieValue,
+        average: Math.floor(hp.hitDieValue / 2) + 1
+      };
+    }
+
+    return advancement;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Enrich all of the entries within the descriptions object on the sheet's system data.
+   * @param {JournalEntryPage} page  Journal page being enriched.
+   * @returns {Promise<object>}      Object with enriched descriptions.
+   */
+  async _getDescriptions(page) {
+    const descriptions = await Promise.all(Object.entries(page.system.description ?? {})
+      .map(async ([id, text]) => {
+        const enriched = await TextEditor.enrichHTML(text, {
+          relativeTo: this.object,
+          secrets: this.object.isOwner,
+          async: true
+        });
+        return [id, enriched];
+      })
+    );
+    return Object.fromEntries(descriptions);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare table based on non-optional GrantItem advancement & ScaleValue advancement.
+   * @param {Item5e} item              Class item belonging to this journal.
+   * @param {number} [initialLevel=1]  Level at which the table begins.
+   * @returns {object}     Prepared table.
+   */
+  async _getTable(item, initialLevel=1) {
+    const hasFeatures = !!item.advancement.byType.ItemGrant;
+    const scaleValues = (item.advancement.byType.ScaleValue ?? []);
+    const spellProgression = await this._getSpellProgression(item);
+
+    const headers = [[{content: game.i18n.localize("DND5E.Level")}]];
+    if ( item.type === "class" ) headers[0].push({content: game.i18n.localize("DND5E.ProficiencyBonus")});
+    if ( hasFeatures ) headers[0].push({content: game.i18n.localize("DND5E.Features")});
+    headers[0].push(...scaleValues.map(a => ({content: a.title})));
+    if ( spellProgression ) {
+      if ( spellProgression.headers.length > 1 ) {
+        headers[0].forEach(h => h.rowSpan = 2);
+        headers[0].push(...spellProgression.headers[0]);
+        headers[1] = spellProgression.headers[1];
+      } else {
+        headers[0].push(...spellProgression.headers[0]);
+      }
+    }
+
+    const cols = [{ class: "level", span: 1 }];
+    if ( item.type === "class" ) cols.push({class: "prof", span: 1});
+    if ( hasFeatures ) cols.push({class: "features", span: 1});
+    if ( scaleValues.length ) cols.push({class: "scale", span: scaleValues.length});
+    if ( spellProgression ) cols.push(...spellProgression.cols);
+
+    const makeLink = uuid => TextEditor._createContentLink(["", "UUID", uuid]).outerHTML;
```

Comment:
I think the synchronous version of this will be going away eventually too. But we might prefer to do this instead:

```suggestion
    const makeLink = uuid => (await fromUuid(uuid))?.toAnchor({classes: ["content-link"]}).outerHTML;
```

The `classes: ["content-link"]` should eventually be unnecessary when we've fixed this small oversight in core.